### ISSUE TITLE
false positive on unused_parens lint

### ISSUE THREAD

**Issue by tomjakubowski** [rust-lang/rust#26461]
``` rust
fn main() {
    if (return) {}
}
```

sez:

```
<anon>:2:8: 2:16 warning: unnecessary parentheses around `if` condition, #[warn(unused_parens)] on by default
<anon>:2     if (return) {}
                ^~~~~~~~
```

However, removing the parentheses makes the file have illegal syntax, because `{}` is part of the return expression (so the `if` is missing its arm).

**Comment by tomjakubowski** [comment]
Also, by the way, it's very amusing to me that this is legal:

```rust
fn main() {
    if return { return } { return } else { return }
}
```

**Comment by samdoiron** [comment]
Triage: Warning still generated in nightly build. Seems to be an issue with [`contains_exterior_struct_lit`](https://github.com/rust-lang/rust/blob/9303055f37a34adfceec1d1012b87308136295ce/src/librustc_lint/unused.rs#L319) not identifying `return {}` as an exterior expression.

**Comment by euclio** [comment]
Looks like this was fixed in #55166, and there's a test.
